lib/activerecord-multi-tenant/migrations.rb

```diff
@@ -82,3 +80,38 @@ def create_table(table_name, options = {}, &block)
     end
   end
 end
+
+module ActiveRecord
+  class SchemaDumper
+    private
+
+    alias initialize_without_citus initialize
+    def initialize(connection, options = {})
+      initialize_without_citus(connection, options)
+
+      @distribution_columns =
+        if ActiveRecord::Migration.citus_version.present?
+          @connection.execute('SELECT logicalrelid::regclass AS table_name, column_to_column_name(logicalrelid, partkey) AS dist_col_name FROM pg_dist_partition').reduce({}) do |h, v|
+            h.merge({ v['table_name'] => v['dist_col_name'] })
+          end
+        else
+          {}
+        end
+    end
+
+    # Support for create_distributed_table & create_reference_table
+    alias table_without_citus table
+    def table(table, stream)
+      table_without_citus(table, stream)
+      table_name = remove_prefix_and_suffix(table)
+      distribution_column = @distribution_columns[table_name]
+      if distribution_column
+        stream.puts "  create_distributed_table(#{table_name.inspect}, #{distribution_column.inspect})"
```

Comment:
Is table_name a string? `#inspect` seems an odd method to escape SQL

Reply:
Yes it is a string AFAIK and agree that it is an odd method for getting escaped quotes in
But used inspect since that is what is used by rails [SchemaDumper](https://github.com/rails/rails/blob/b7eaebe5a7da0a8bceead0cabb790d6e40c1a396/activerecord/lib/active_record/schema_dumper.rb#L136) even from its [first commit](https://github.com/rails/rails/blob/7dc45818dc43c163700efc9896a0f3feafa31138/activerecord/lib/active_record/schema_dumper.rb)